practical-react-unit-testing-with-jest-mocks-and-spies/Exercise1/problem/src/Select.test.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * @jest-environment jsdom
+ */
+//manually mocking toSelectOptions
+
+import { render } from "@testing-library/react";
+import { Select } from "./Select";
+import { toSelectOptions } from "./toSelectOptions";
+
+// mock the toSelectOptions module with a callback that returns a mock module
+//   mocking the toSelectOptions function
+
+describe("Year", () => {
+  test("Year works with one-off mocked values", () => {
+    // Use mockImplementationOnce to get a 2024 value for just this one render
+    const result = render(
+      <Select
+        label="Year"
+        name="year"
+        value="2024"
```

Comment:
The problem doesn’t line up with the solution in some places, like the values. I would get the solution working as you want and then copy it to the problem, and remove the parts you want the participants to complete.